src/passes/OptimizeInstructions.cpp

```diff
@@ -298,15 +298,18 @@ struct OptimizeInstructions
         }
       }
       {
-        // eqz((signed)x % C_pot)  =>  eqz(x & (C_pot - 1))
+        // eqz((signed)x % C_pot)  =>  eqz(x & (abs(C_pot) - 1))
+        // eqz((signed)x % (i32|i64).min_s)  =>  eqz(x & (i32|i64).max_s)
         Const* c;
         Binary* inner;
         if (matches(curr,
                     unary(Abstract::EqZ,
                           binary(&inner, Abstract::RemS, any(), ival(&c)))) &&
-            Bits::isPowerOf2((uint64_t)c->value.getInteger())) {
+            (Bits::isPowerOf2(c->value.abs().getInteger()) ||
+             c->value.getUnsigned() ==
+               (1ULL << (c->type.getByteSize() * 8 - 1)))) {
           inner->op = Abstract::getBinary(c->type, Abstract::And);
-          c->value = c->value.sub(Literal::makeFromInt32(1, c->type));
+          c->value = c->value.abs().sub(Literal::makeFromInt32(1, c->type));
```

Comment:
If the value is INT_MIN, calling `abs()` is undefined behavior, so I think that needs to be handled specially.

Reply:
done!

Reply:
What's funny, running from UB got UB =)